src/wallet/spend.cpp

```diff
@@ -575,6 +584,14 @@ std::optional<SelectionResult> SelectCoins(const CWallet& wallet, CoinsResult& a
         return result;
     }
 
+    // Return early if we cannot cover the target with the wallet's UTXO.
+    // We use the total effective value if we are not subtracting fee from outputs and 'available_coins' contains the data.
+    CAmount available_coins_total_amount = coin_selection_params.m_subtract_fee_outputs ? available_coins.total_amount :
+            (available_coins.total_effective_amount.has_value() ? *available_coins.total_effective_amount : available_coins.total_amount);
```

Comment:
I don't think it's correct to fallback on `available_coins.total_amount` when we don't do SFFO.

with SFFO we must have effective amount, if we don't - something went wrong.

Reply:
Hmm, need a bit of context here.
Guess that you are saying that because, with SFFO, the wallet needs to select a group of UTXOs that will be able to pay for themselves? So, the entire coin selection relies on the effective value existence so.. if it's not there there is a bug?

Reply:
By the time we get here, if `total_effective_amount` doesn't have a value, that means that none of the UTXOs in this wallet were solvable, so we can't do any coin selection here in that case. So I think it would be better to always return early in that case.

Reply:
> By the time we get here, if total_effective_amount doesn't have a value, that means that none of the UTXOs in this wallet were solvable, so we can't do any coin selection here in that case

Hmm, code wise, that doesn't seems to be what is happening.

At the `COutput` constructor, if the coin isn't solvable then `effective_value` will be equal to the nominal output value. So it will always have an effective value during the tx creation process.

The only way that a `COutput` has no `effective_value` is when we don't provide the feerate param to `AvailableCoins` (which happens on all the flows that uses `AvailableCoins` that are not part of the tx creation process).

Reply:
Oh, hmm, I thought it checked `input_bytes` as well.

In that case, `total_effective_amount` should always have a value if we have any coins. If it isn't set, then we don't have any coins and we can still return early.